bin/hostname/tests/hostname_test.sh

```diff
@@ -0,0 +1,57 @@
+#-
+# Copyright (c) 2023 Klara, Inc.
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+#
+# These tests need to run in a multibyte locale with non-localized
+# error messages.
+#
+export LC_CTYPE=C.UTF-8
+export LC_MESSAGES=C
+
+root_dir=$(atf_get_srcdir)
+
+test_jail_conf='test-hostname-jail {
+    host.hostname = "test-hostname";
+    path = "/";
+    persist;
+}'
+
+init()
+{
+    echo "$test_jail_conf" > "$root_dir/jail.conf"
+    jail -f "$root_dir/jail.conf" -c test-hostname-jail
+}
+
+recycle()
+{
+    jail -f "$root_dir/jail.conf" -r test-hostname-jail
+    rm "$root_dir/jail.conf"
+}
+
+atf_test_case basic cleanup
+basic_head()
+{
+    atf_set require.user root
+    atf_set "descr" "main test"
+}
+basic_body()
+{
+    init
+    jexec test-hostname-jail "hostname"
+    result=$(jexec test-hostname-jail "hostname")
+    if [ "$result" != "test-hostname" ]; then
+        atf_fail "Hostname does not match: $result" 
+    fi
```

Comment:
We can use `atf_check_equal "test-hostname" "${result}"` to replace these 3 lines.